Extensions.cs

```diff
@@ -216,5 +217,14 @@ public static Color SafeColorRange(this Color valuecolor)
         {
             return new Color(Mathf.Clamp(valuecolor.r, 1f / 255f, 1f), Mathf.Clamp(valuecolor.g, 1f / 255f, 1f), Mathf.Clamp(valuecolor.b, 1f / 255f, 1f));
         }
+
+        public static ILCursor GotoNextOutEnd(this ILCursor cursor, out ILLabel end, params Func<Mono.Cecil.Cil.Instruction, bool>[] predicates)
+        {
+            var ret = cursor.GotoNext(moveType: MoveType.AfterLabel, predicates);
+            var c = cursor.Clone();
+            c.Index += predicates.Length;
+            end = c.MarkLabel();
+            return ret;
+        }
```

Comment:
🤨 

I see the attempt at making a pattern out of these types of matches, but 1. it's not a lot of code; 2. they'll all have small nuances specially with how if/else blocks might be organized; and lastly 3. I wouldn't encourage matching the whole body of an if, matches are ideally as short as possible so to not be affected by other mods' edits